### PR TITLE
Add data-brand attribute to Brand Component

### LINKED ISSUE #1853: Event tracking - data attributes for brand component
**Is your feature request related to a problem? Please describe.**
Add data attributes for event tracking in the brand component.

**Describe the solution you'd like**
Add data attributes to the brand component which will be used for tracking the component in Simorgh, e.g. `data-brand=header`.

Data attributes should be added to the brand in the header and the footer, i.e. `data-brand=header`, `data-brand=footer`.

**Testing notes**
[Tester to complete]

Dev insight: Snapshots

- [ ] This feature is expected to need manual testing.

**Additional context**
* https://paper.dropbox.com/doc/Event-tracking--AjZNWSURmJ_~rHV8QUSJkDShAg-3i47TvVb9IJBMJFcL8D6u#:uid=275080311694076762638564&h2=Brand
* WSFRONTPGS-178
* https://github.com/bbc/simorgh/tree/latest/src/app/containers/ATIAnalytics

### PR DESCRIPTION
**Pull request by ghost**
Resolves #1853

**Overall change:** _Add `data-brand` attribute to brand to be used during event tracking._

**Code changes:**

- _Add `data-brand` data attribute to be used for event tracking._
- _Add `location` prop to identify where the brand is rendered._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by DenisHdz — APPROVED** [review]
LGTM

**Review by HarveyPeachey — COMMENTED** [review]
LGTM. But does the `location` prop need to be added to the storybook example?

File: packages/components/psammead-brand/src/index.jsx [hunk 1]
```diff
@@ -212,6 +215,7 @@ Brand.propTypes = {
   serviceLocalisedName: string,
   borderTop: bool,
   borderBottom: bool,
+  location: string,
```
Comment (OlgaLyubin) [inline packages/components/psammead-brand/src/index.jsx:218]: Should this not be `oneOf(['header', 'footer'])` ?

Comment (ghost) [reply packages/components/psammead-brand/src/index.jsx:218]: I left it open to other possible values considering that location could be anywhere on the page.

**Comment by ghost** [comment]
@HarveyPeachey I didn't add location to the storybook knobs because it doesn't trigger a visual change to the component. Adding it there would be pointless IMO.

**Review by HarveyPeachey — APPROVED** [review]

File: packages/components/psammead-brand/src/index.jsx (hunk 1 above)
Comment (OlgaLyubin) [reply packages/components/psammead-brand/src/index.jsx:218]: Okay, makes sense

**Review by OlgaLyubin — APPROVED** [review]
Looks good

**Review by dr3 — CHANGES_REQUESTED** [review]
Rather than having custom logic in psammead for simorgh event tracking, can we instead spread props into this component, and set the data attribute in simorgh please?  IMO we shouldnt have custom logic specific to simorgh when its not needed. Someone else using Brand may not do event tracking like we are. It also makes it very difficult for us to change our event tracking method in the future.

**Comment by j-pendlebury** [comment]
agreed with @dr3. Could you make the changes necessary in this PR and I'll write up issues for fixing what's already been merged and the Simorgh side of things

**Comment by ghost** [comment]
@dr3 Good call.

**Comment by ghost** [comment]
Closed in favour of https://github.com/bbc/psammead/pull/2377

**Pull request closed without merging**